--- Base.py.orginial	2012-04-28 19:24:30.000000000 +0000
+++ Base.py	2014-03-14 07:01:17.513173400 +0000
@@ -282,6 +282,11 @@
                 # servers worked:
                 first_socket_error = first_socket_error or e
                 continue
+            except TimeoutError as t:
+                first_socket_error = first_socket_error or t
+                continue
+            if self.response:
+                break
         if not self.response and first_socket_error:
             raise first_socket_error
 
@@ -319,6 +324,11 @@
             except socket.error as e:
                 first_socket_error = first_socket_error or e
                 continue
+            except TimeoutError as t:
+                first_socket_error = first_socket_error or t
+                continue
+            if self.response:
+                break
         if not self.response and first_socket_error:
             raise first_socket_error
 
